sdcm/nemesis.py

```diff
@@ -2532,22 +2532,45 @@ def _corrupt_data_file(self):
             self.log.debug('File {} was corrupted by dd'.format(sstable_file))
 
-    def disable_disrupt_corrupt_then_scrub(self):
-
-        # Flush data to sstable
-        self.target_node.run_nodetool("flush")
-
-        self.log.debug("Randomly corrupt sstable file")
-        self._corrupt_data_file()
-
-        self.log.debug("Rebuild sstable file by scrub, corrupted data file will be skipped.")
-        for ks_cf in self.cluster.get_non_system_ks_cf_list(self.target_node):
-            self.target_node.run_nodetool("scrub", args=f"{ks_cf.split('.')[0]}")
-
-        self.log.debug(
-            'Refreshing the cache by restart the node, and verify the rebuild sstable can be loaded successfully.')
-        self.target_node.stop_scylla_server(verify_up=False, verify_down=True)
-        self.target_node.start_scylla_server(verify_up=True, verify_down=False)
+    def disrupt_corrupt_then_scrub(self):
+        """
+        Load sstables with invalid fragment by refresh, the sstables were generated by
+        scylla unittest (test/boost/sstable_datafile_test.cc:sstable_scrub_test).
+        Then try rebuild the sstables by scrub, the invalid sstable will be skipped if
+        `--skip-corrupted` option is used.
+        """
+        # Create test ks and cf before refresh sstables
+        node = self.target_node
+        with self.cluster.cql_connection_patient(node) as session:
+            session.execute("""
+                CREATE KEYSPACE IF NOT EXISTS ks
+                WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '3'} AND durable_writes = true;
+                """)
+            session.execute("""
+                CREATE TABLE IF NOT EXISTS ks.cf (
+                    pk text,
+                    ck int,
+                    s int,
+                    v int,
+                    PRIMARY KEY (pk, ck));
+                """)
+        node.run_nodetool("flush")
+
+        # Copy invalid sstables to upload directory of target node, and load the sstables by refresh
+        result = node.remoter.sudo("ls -t /var/lib/scylla/data/ks/")
+        cf_dir = result.stdout.split()[0]
+        upload_dir = os.path.join('/var/lib/scylla/data/ks/', cf_dir, 'upload')
+        node.remoter.send_files(src='data_dir/sstable_with_invalid_fragment', dst='/tmp')
+        node.remoter.sudo(f'mv /tmp/sstable_with_invalid_fragment/ks/cf-test/* {upload_dir}')
+        node.remoter.sudo(f'sudo chown -R scylla:scylla {upload_dir}')
+        node.run_nodetool(sub_cmd="refresh", args="-- ks cf")
+
+        self.log.debug("Rebuild sstable file by scrub, scrub will be stop for invalid sstable.")
+        node.run_nodetool("scrub", args="ks cf")
+
+        self.log.debug("Rebuild sstable file by scrub with `--skip-corrupted`, corrupted data file will be skipped.")
+        with ignore_scrub_invalid_errors():
+            node.run_nodetool("scrub", args="--skip-corrupted ks cf")
```

Comment:
@amoskong I don't see a point in repeating this if we already have the same thing in dtest.
The idea in SCT is not to load sstables and change them, but to cause a corruption (if possible) to the sstables that are part of the test.
If it's not possible to use the same method of corrupting the sstables like in the unit test so the nodetool scrub work, we can close this PR and not have it in SCT.

Reply:
How about just run `nodetool scrub` and `nodetool scrub --skip-corrupted` for existing test tables? the tables contains a lot of data (sstables). We expect no crash, no special error

Reply:
ok.
run scrub but without the --skip-corrupted.